worker/vcs/manager.go

```diff
@@ -176,23 +196,24 @@ func (vm *Manager) CloneVersionRepository(event *api.Event) error {
 		}
 	} else {
 		// checkout special commit
-		if err = worker.CheckOutByCommitID(event.Version.Commit, destPath, event); err != nil {
+		if err := worker.CheckOutByCommitID(event.Version.Commit, destPath, event); err != nil {
 			event.Service.Repository.Status = api.RepositoryMissing
 			steplog.InsertStepLog(event, steplog.CloneRepository, steplog.Stop, err)
 			return fmt.Errorf("Unable to check out commit %s :%v\n", event.Version.Commit, err)
 		}
 	}
 
-	if api.APIOperator == event.Version.Operator {
-		// create tag
-		if err := worker.NewTagFromLatest(destPath, event); err != nil {
-			log.Errorf("Unable to push new commit %s :%v\n", event.Version.Commit, err)
-		}
-	}
-	steplog.InsertStepLog(event, steplog.CloneRepository, steplog.Finish, nil)
 	return nil
 }
 
+// formatVersionName replace the random name with default name '$createTime|$commitID' when name empty in create version
+func (vm *Manager) formatVersionName(event *api.Event) {
+	if bson.IsObjectIdHex(event.Version.Name) && event.Version.Commit != "" {
```

Comment:
If the version name is empty, there is no need to assign it an `ObjectIdHex` in the front. Just leave it empty, and assign it the correct value at the right place.